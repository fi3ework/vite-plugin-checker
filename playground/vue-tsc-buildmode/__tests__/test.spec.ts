import stringify from 'fast-json-stable-stringify'
import { describe, expect, it } from 'vitest'

import {
  diagnostics,
  editFile,
  expectStderrContains,
  isBuild,
  isServe,
  log,
  resetReceivedLog,
  sleepForEdit,
  sleepForServerReady,
  stripedLog,
} from '../../testUtils'

describe('vue-tsc-build-mode', () => {
  describe.runIf(isServe)('serve', () => {
    it('get initial error and subsequent error', async () => {
      await sleepForServerReady(2)
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()

      console.log('-- edit file --')
      resetReceivedLog()
      editFile('packages/utils/src/helpers.ts', (code) =>
        code.replace('processData(input: number)', 'processData(input: boolean)')
      )
      await sleepForEdit(2)
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()
    })
  })

  describe.runIf(isBuild)('build', () => {
    it('should fail', async () => {
      const expectedMsg = `error TS2345`
      expectStderrContains(log, expectedMsg)
    })
  })
})
