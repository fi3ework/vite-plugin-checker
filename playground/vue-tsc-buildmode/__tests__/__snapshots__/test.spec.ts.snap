// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vue-tsc-build-mode > serve > get initial error and subsequent error 1`] = `"[{"checkerId":"vue-tsc","frame":"    2 |/n    3 | export function processData(data: string): string {/n  > 4 |   const processed = utilsProcessData(data);/n      |                                      ^^^^/n    5 |   return formatMessage(capitalize(processed));/n    6 | }","id":"<PROJECT_ROOT>/playground-temp/vue-tsc-buildmode/packages/core/src/index.ts","level":1,"loc":{"column":38,"file":"<PROJECT_ROOT>/playground-temp/vue-tsc-buildmode/packages/core/src/index.ts","line":4},"message":"Argument of type 'string' is not assignable to parameter of type 'number'.","stack":""}]"`;

exports[`vue-tsc-build-mode > serve > get initial error and subsequent error 2`] = `
[
  "
 ERROR(vue-tsc)  Argument of type 'string' is not assignable to parameter of type 'number'.
 FILE  <PROJECT_ROOT>/playground-temp/vue-tsc-buildmode/packages/core/src/index.ts:4:38

    2 |
    3 | export function processData(data: string): string {
  > 4 |   const processed = utilsProcessData(data);
      |                                      ^^^^
    5 |   return formatMessage(capitalize(processed));
    6 | }

[vue-tsc] Found 1 error. Watching for file changes.",
]
`;

exports[`vue-tsc-build-mode > serve > get initial error and subsequent error 3`] = `"[{"checkerId":"vue-tsc","frame":"    2 |/n    3 | export function processData(data: string): string {/n  > 4 |   const processed = utilsProcessData(data);/n      |                                      ^^^^/n    5 |   return formatMessage(capitalize(processed));/n    6 | }","id":"<PROJECT_ROOT>/playground-temp/vue-tsc-buildmode/packages/core/src/index.ts","level":1,"loc":{"column":38,"file":"<PROJECT_ROOT>/playground-temp/vue-tsc-buildmode/packages/core/src/index.ts","line":4},"message":"Argument of type 'string' is not assignable to parameter of type 'number'.","stack":""},{"checkerId":"vue-tsc","frame":"    2 |/n    3 | export function processData(data: string): string {/n  > 4 |   const processed = utilsProcessData(data);/n      |                                      ^^^^/n    5 |   return formatMessage(capitalize(processed));/n    6 | }","id":"<PROJECT_ROOT>/playground-temp/vue-tsc-buildmode/packages/core/src/index.ts","level":1,"loc":{"column":38,"file":"<PROJECT_ROOT>/playground-temp/vue-tsc-buildmode/packages/core/src/index.ts","line":4},"message":"Argument of type 'string' is not assignable to parameter of type 'boolean'.","stack":""}]"`;

exports[`vue-tsc-build-mode > serve > get initial error and subsequent error 4`] = `
[
  "
 ERROR(vue-tsc)  Argument of type 'string' is not assignable to parameter of type 'boolean'.
 FILE  <PROJECT_ROOT>/playground-temp/vue-tsc-buildmode/packages/core/src/index.ts:4:38

    2 |
    3 | export function processData(data: string): string {
  > 4 |   const processed = utilsProcessData(data);
      |                                      ^^^^
    5 |   return formatMessage(capitalize(processed));
    6 | }

[vue-tsc] Found 1 error. Watching for file changes.",
]
`;
